plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -26,6 +29,93 @@
 
 import time
 import agate
+import re
+
+
+BQ_INTEGER_RANGE_NOT_SUPPORTED = f"""
+BigQuery integer range partitioning is only supported by the
+`partition_by` config, which accepts a dictionary.
+
+See: {dbt.links.BigQueryNewPartitionBy}
+"""
+
+
+@dataclass
+class PartitionConfig(Dict[str, Any], JsonSchemaMixin):
```

Comment:
If I understand what's going on here correctly, I think instead of inheriting this from `Dict` you should pass `partition_by.to_dict()` instead of the places you're using this as a dict.